crates/biome_glob/src/lib.rs

```diff
@@ -1,31 +1,86 @@
-use biome_rowan::{TextRange, TextSize};
+//! biome_glob provides a glob and glob list with exceptions matching.
+//!
+//! ## Matching a path against a glob
+//!
+//! You can create a glob from a string using [core::str::FromStr::from_str] or the corresponding method `parse`.
+//! A glob can match against anything that can be turned into a [std::path::Path].
+//! This is for example the aqcse of strings.
+//!
+//! ```
+//! use biome_glob::Glob;
+//!
+//! let glob = "*.rs".parse::<Glob>().expect("correct glob");
+//! assert!(glob.is_match("lib.rs"));
+//! assert!(!glob.is_match("src/lib.rs"));
```

Comment:
This example doesn't show the usage of `FromStr::from_str` and we should add it.

Also, I would remove the `is_match` because the previous paragraph didn't mention the `is_match` function. If you intend the keep `is_match` in the example, you should explain it in the previous paragraph, and explain why `src/lib.rs` isn't a match, and why `lib.rs` is a match.